src/main/scala/com/mwz/sonar/scala/util/util.scala

```diff
@@ -19,56 +19,38 @@
 package com.mwz.sonar.scala
 package util
 
-import java.io.File
 import java.nio.file.{Path, Paths}
 import java.util.Optional
 
-import org.sonar.api.batch.fs.{FileSystem, InputFile}
-import org.sonar.api.batch.measure.Metric
-import org.sonar.api.batch.sensor.SensorContext
-import org.sonar.api.config.Configuration
+import org.sonar.api.batch.fs.FileSystem
 
-import scala.language.{higherKinds, implicitConversions}
-import scala.util.Try
+import scala.language.implicitConversions
 
 /**
  *  Scala.Option <-> Java.Optional conversions.
  *  @note Taken from https://gist.github.com/julienroubieu/fbb7e1467ab44203a09f.
  */
 object JavaOptionals {
-  implicit def toRichOption[T >: Null](opt: Option[T]): RichOption[T] = new RichOption[T](opt)
-  implicit def toRichOptional[T](optional: Optional[T]): RichOptional[T] = new RichOptional[T](optional)
+  implicit final def toRichOption[T >: Null](opt: Option[T]): RichOption[T] =
+    new RichOption[T](opt)
+  implicit final def toRichOptional[T](optional: Optional[T]): RichOptional[T] =
+    new RichOptional[T](optional)
 }
 
 /** Transform this Option to an equivalent Java Optional */
-class RichOption[T >: Null](opt: Option[T]) {
+final class RichOption[T >: Null](val opt: Option[T]) extends AnyVal {
```

Comment:
:+1: for making this a value class - _(I believe I was the one that first coded this, so thanks for improving it)_
And again, why don't make this an implicit class instead of using implicit conversions?

Also, I think this counts as syntax too. Thus, we should move them to the syntax `object/package` and rename them **OptionOps** and **OptionalOps** just to be consistent.

Reply:
> 👍 for making this a value class - _(I believe I was the one that first coded this, so thanks for improving it)_

👍 

I think that I addressed the first part of your comment above, but regarding those option conversions I didn't want to refactor too much at once and tried to touch as little as possible that affects other modules, so, for now, I might only add a TODO to refactor this later or possibly in a separate PR.